server/controllers/shipment_cost.js

```diff
@@ -94,26 +97,74 @@ function addRules (rules) {
   }
   return engine
 }
-
 function runRules (engine, facts, res) {
   let array
-  engine.run(facts).then(triggeredEvents => {
-    // engine returns a list of events with truthy conditions
-    let cost = { cost: 0 }
-    array = triggeredEvents.map(event => (formula(event, facts, cost)))
-    res.status(httpStatus.OK)
-    res.send(getResult(array, cost.cost))
-  }).catch((err) => res.send({ message: 'ERROR: ' + err }))
+  engine
+    .run(facts)
+    .then(triggeredEvents => {
+      // engine returns a list of events with truthy conditions
+      let cost = { cost: 0 }
+      array = triggeredEvents.map(event => (formula(event, facts, cost)))
+      if (array.length === 0) array = ['disabled']
+      res
+        .status(httpStatus.OK)
+        .json({ success: true, cost: getResult(array, cost.cost) })
+    })
+    .catch((error) => {
+      res
+        .status(httpStatus.INTERNAL_SERVER_ERROR)
+        .json({ success: false, errors: error })
+    })
+}
+
+function isDictionary (obj) {
```

Comment:
primero, se llaman objetos acá, no diccionarios
segundo, hay formas más cortas: https://stackoverflow.com/a/8511350

Reply:
Puse dictionary porque in array tambien es un objeto